src/client/mod.rs

```diff
@@ -143,8 +141,6 @@ impl ClientBuilder {
 
     /// Sets the application id.
     pub fn application_id(mut self, application_id: u64) -> Self {
-        self.application_id = Some(ApplicationId(application_id));
-
         if let Some(http) = &mut self.http {
```

Comment:
`self` doesn't need to be mutable anymore now:

```suggestion
    pub fn application_id(self, application_id: u64) -> Self {
        if let Some(http) = &self.http {
```